src/main/java/apoc/nodes/Nodes.java

```diff
@@ -303,6 +304,26 @@ public boolean isDense(@Name("node") Node node) {
         }
     }
 
+    @Procedure(mode = Mode.WRITE)
+    @Description("apoc.nodes.removewithlabels(['Person','Movie'],1000) | Will remove all the nodes having one of the given labels, via DETACH DELETE using apoc.periodic.iterate. Same limit used for each label. Default is 100. ")
+    public Stream<MapResult> removewithlabels(@Name("labels") List<String> labels, @Name(value = "limit",defaultValue="100")  Long limit) {
+        if (limit <= 0)
+            throw new RuntimeException("limit must be a positive integer");
+
+        Map<String,Object> results = new HashMap<String, Object>();
+        HashMap<String, Object> parameters = new HashMap<>();
+
+        for( String label: labels) {
+            if(label.trim().length()>0) {
+                String query = String.format("call apoc.periodic.iterate( 'MATCH (n:`%s`) RETURN n',  'DETACH DELETE n', {batchsize:%d,iterateList:true, parallel:true}  )", label, limit);
```

Comment:
parallel true doesn't work here
best to prefix the statement with `cypher runtime=slotted MATCH (...)`
quote the labels

Reply:
I'm not 100% sure apoc.periodic.iterate works in the same tx. please also test it in a server environment with a lot of highly connected labels (1M nodes 2M rels)

Reply:
I heard slotted is Entreprise only in your video. Unsure how it would behave in community.

Reply:
org.neo4j.graphdb.QueryExecutionException: Failed to invoke procedure  if I use runtime=slotted

As seen in debug:
community single [/home/jerome/OpenSource/wadael/neo4j-apoc-procedures/target/test-data/impermanent-db]